templates/default/filter.conf.erb

```diff
@@ -2,6 +2,6 @@
 # Do NOT modify this file by hand!
 
 <filter <%= @tag %>>
-  type <%= @type %>
+  <%= @node['td_agent']['template_type_string'] %> <%= @type %>
```

Comment:
We're doing similar switch for `@include` based on td-agent's major version.

https://github.com/treasure-data/chef-td-agent/blob/v2.5.0/templates/default/td-agent.conf.erb#L2-L6

AFAIK, we're planning to release fluentd 0.14.x as new major release td-agent 3.x. What if we do this switch based on td-agent's major version?

https://github.com/treasure-data/omnibus-td-agent/blob/v3.0.1/config/projects/td-agent2.rb#L23
https://github.com/treasure-data/omnibus-td-agent/blob/v3.0.1/config/projects/td-agent3.rb#L25

Reply:
BTW, could you split this PR into 2 pieces; `@type` and `install_url`?

Reply:
I'll look into auto-detecting the `type` vs `@type` based on the version but I'd rather allow manual override (as I've often found auto-detection to break in unexpected ways).

Also following your links above something does not look right with the deb version and the td-agent version:
```
$ sudo dpkg --list | grep td-agent
ii  td-agent                         2.3.5-0                           amd64        Treasure Agent: A data collector for Treasure Data
$ td-agent --version
td-agent 0.14.20
```

Should the deb package version 2.3.5 install td-agent 0.14.20?